### ISSUE TITLE
How to use focal images?

### ISSUE THREAD

**Issue by xnox** [canonical/multipass#1417]
focal images already exist in daily streams, but not yet in release streams.

It seems to me, that once they do land in the release stream "multipass will not just work" because of the hardcoded mappings at https://github.com/canonical/multipass/blob/master/src/daemon/custom_image_host.cpp#L56

Could you please update those mappings for core20 / 20.04 / focal pointing at the non-existent release streams? Such that when they will be published there, "multipass just starts working" with them?

It would be also nice to have ability to point multipass at the daily streams, but imho that is a separate request which i have filed at https://github.com/canonical/multipass/issues/1416

**Comment by Saviq** [comment]
@xnox focal images are already available with `daily:focal`, is that not what you're after? If you mean `core20` images, then similar to #1416 we "bless" those images once they becomes stable. There are no streams for Core images that I know of?

**Comment by xnox** [comment]
I'm talking about snapcraft VMs that are used as snap build-environments. Those are classic VMs use buildd variant. Currently there is only a daily stream for buildds. From https://cloud-images.ubuntu.com/buildd/daily/streams/v1/com.ubuntu.cloud:daily:download.json I think.

It would be nice if you didn't hardcode blessings. For example, you can be sensitive to the "supported: true|false" key in the stream. Currently focal has it set to false, whilst others are set to true. And I would want to have a way to "launch" unblessed images, or self bless them myself.

Also, why are we having mappings in the multipass code, do you want more aliases on the images? Cause for example, in the buildd streams we can add alias "core20" on the focal image, and "core18" on the bionic one, etc.

**Comment by Saviq** [comment]
@xnox there are no other images than `focal` in the buildd stream (yet, that I know of) that Multipass can consume (there are no VM images, only LXD ones). We'll definitely get on that bandwagon when they're up.

As for those mappings, snapcraft basically calls `multipass launch snapcraft:$base`, as long as there are only `core*` bases, we could just point at, a suitably populated, buildd stream. But we're going to have bases that won't live in that stream - so there has to be a mapping somewhere. It may be more natural to go into snapcraft, like it did for the LXD backend.

**Comment by xnox** [comment]
/me goes to see where/what/why buildd VM images are not up for focal.

**Comment by xnox** [comment]
Ergh, they are there....

```
      "disk1.img": {
       "ftype": "disk1.img", 
       "path": "focal/20200313/focal-server-cloudimg-amd64-disk.img", 
```
https://cloud-images.ubuntu.com/buildd/daily/focal/current/

has https://cloud-images.ubuntu.com/buildd/daily/focal/current/focal-server-cloudimg-amd64-disk.img

Is that not what multipass/snapcraft/core20 VMs need ?

**Comment by xnox** [comment]
Maybe i should just wait, cause it seems like they are not there for other series yet.

**Comment by Saviq** [comment]
@xnox yes, they're there for focal, and you can use Multipass from #1192 (`snap refresh multipass --channel edge/pr1192`) to get at that.

**Comment by Saviq** [comment]
(edited above to have the right PR number)

**Comment by xnox** [comment]
@Saviq when will that land? It looks like multipass is the last one that doesn't support core20 yet.

```
git clone -b 20 https://git.launchpad.net/~canonical-foundations/snap-pc/+git/github-mirror-amd64
snapcraft
```
Fails. With that PR it starts up the VM.

Yet, creating a /+newsnap for it in launchpad works with stock/default settings, no tricks, no hacks. https://launchpad.net/~xnox/+snap/test-gadget-base-core20

And running `snapcraft --use-lxd` locally also works.

Ubuntu 20.04 LTS is now out, and all the buildd images and core20 are now stable, and have security support.

**Comment by Saviq** [comment]
Hi @xnox, the buildd images are not considered stable, because there's no CI for them (to test what they're meant to be doing), so there's no way to automatically promote from `daily` to `release`.

AFAICT, [`core20`](http://cdimage.ubuntu.com/ubuntu-core/20/) isn't stable either?

I've now proposed #1520 to add the minimal 20.04 image as `core20` builder - the same that we use for `core16` and `core20`.
